api/src/main/java/org/openmrs/validator/AllergyValidator.java

```diff
@@ -87,6 +87,12 @@ public void validate(Object target, Errors errors) {
 					errors.rejectValue("allergen", "allergyapi.message.duplicateAllergen", new Object[] { name }, null);
 				}
 			}
+			
+			if(StringUtils.isNotBlank(allergen.getNonCodedAllergen())){
+				if (!allergen.getNonCodedAllergen().matches("[a-zA-Z]+$")){
+					errors.rejectValue("allergen", "error.allergyapi.allergen.nonCodedAllergen.validateInputString");
```

Comment:
This looks nice and good progress, Replace this validator message with **Other Allergen must not contain a number or symbol** the one you have  globally called

Reply:
@sherrif10  according to @mozzy11 ..I was told its better to internationalize ... which this message "error.allergyapi.allergen.nonCodedAllergen.validateInputString" represents in the message.properties file
Hope this was already resolve

Reply:
error.allergyapi.allergen.nonCodedAllergen.validateInputString=Other Allergen must not contain a number or symbol

.........Has u see up that is in the message.properties